packages/flutter/lib/src/material/snack_bar.dart

```diff
@@ -424,3 +423,58 @@ class SnackBar extends StatelessWidget {
     );
   }
 }
+
+class _SnackbarSemantics extends SingleChildRenderObjectWidget {
+  const _SnackbarSemantics({
+    @required Widget child,
+    @required this.onDismiss,
+  }) : assert(onDismiss != null),
+       super(child: child);
+
+  final VoidCallback onDismiss;
+
+  @override
+  RenderObject createRenderObject(BuildContext context) {
+    return _RenderSnackbarSemantics()
+      ..textDirection = Directionality.of(context)
+      ..onDismiss = onDismiss;
+  }
+
+  @override
+  void updateRenderObject(BuildContext context, covariant _RenderSnackbarSemantics renderObject) {
+    renderObject
+      ..textDirection = Directionality.of(context)
+      ..onDismiss = onDismiss;
+  }
+}
+
+class _RenderSnackbarSemantics extends RenderProxyBox {
+  String _lastAnnouncedLabel;
+
+  TextDirection textDirection;
+
+  VoidCallback onDismiss;
+
+  void _onDismiss() {
+    if (onDismiss != null) {
+      onDismiss();
+    }
+  }
+
+  @override
+  void describeSemanticsConfiguration(SemanticsConfiguration config) {
+    super.describeSemanticsConfiguration(config);
+    config.liveRegion = true;
+    config.isSemanticBoundary = true;
+    config.onDismiss = _onDismiss;
+  }
+
+  @override
+  void assembleSemanticsNode(SemanticsNode node, SemanticsConfiguration config, Iterable<SemanticsNode> children) {
+    super.assembleSemanticsNode(node, config, children);
+    if (defaultTargetPlatform == TargetPlatform.iOS && node.label != null && node.label != _lastAnnouncedLabel) {
+      _lastAnnouncedLabel = node.label;
+      SemanticsService.announce(_lastAnnouncedLabel, textDirection);
```

Comment:
This seems hacky to make an announcement as a side effect of assembling the tree...

Reply:
Yes, its a bit of a hack. Unfortunately since we take a `child: Widget` instead of a text String I don't think we have a better way of figuring out what to announce until after we've built the semantics tree. We could build a live region concept into our own semantics...